main/RFXtrx.h

```diff
@@ -2242,6 +2242,7 @@ typedef union tRBUF {
 		BYTE	battery_level : 4;
 		BYTE	rssi : 4;
 #endif
+		BYTE	id0;
```

Comment:
Please do not modify this file. It is read-only and copyrighted

Reply:
Ok, that is a bit more work. I will work on it. The first 2 reviews are now covered in my commit 5 minutes ago.

Reply:
Thank you. Still the odds that you have a device that have the same 16 bits is very slim.

Reply:
I agree, but lets make it right for once and forever.

Reply:
I don't share the same idea... 
- It is working currently for everyone, and no one will have multiple rain devices, and if they do, the chance that it has the same 16 bits is the same as winning the big lottery ticket
- You are trying to squeeze in bits that may be there in RTL-433, but not on other RF hardware that is mainly used inside the application
- For every received device (RTL433) a extra string compare is performed (slowing down the application, not by much, but still)

If the battery is being correctly with the below, then that would be a great PR

`decode_BateryLevel(subType == sTypeRAIN1, pResponse->RAIN.battery_level & 0x0F);`

Reply:
So I propose to leave the structure alone and use the 16 bits of the 20 instead of just 8. That will avoid the conflict I noticed with minimal change. By the way I also prefer RFXtrx but for the rain sensor it is current less stable. Devices are lost and I dont know the reason but rtl-sdr works all the time without loosing devices. That is why I use rtl-sdr and run rfxtrx in parrallel to help RFXcom debug/test. By the way I have 5 rain sensors. Two for master rain to determine if I need to rain the garden and 3 more to monitor the groups of the rain system.

Reply:
We are using RFXCom RFXtrx433 (XL) for many many years without any issues... Maybe you enabled to many protocols. Some protocols have influence on others.
It is also a transceiver instead of only a receiver.

You could have rebased this PR , but I can also squash it into a single commit.

Okey so the only change will be the battery ?

Reply:
I agree the RFXtrx433E and XL are great devices. They work fine in most cases. It just for the TFA rain sensor where the SDR works better. I run the RFXtrx and the rtlsdr in parrallel for debugging. For the first 2 days or so both the RFXtrx and RTL-SDR reports all 3 TFS Raindrops perfect. At some pint the RFXtrx stops reporting the Raindrop while the RTL-SDR contains as expected. I have no clue what it is. For other devices the RFXtrx433 continues to function normal. That is why I currently use the RTL-SDR with Raindrop on my active system. Thats all